apps/tangle-dapp/src/pages/bridge/containers/BridgeContainer.tsx

```diff
@@ -0,0 +1,848 @@
+    (state) => state.isAmountInputError,
+  );
+  const setIsAmountInputError = useBridgeStore(
+    (state) => state.setIsAmountInputError,
+  );
+  const amountInputErrorMessage = useBridgeStore(
+    (state) => state.amountInputErrorMessage,
+  );
+  const destinationAddress = useBridgeStore(
+    (state) => state.destinationAddress,
+  );
+  const setDestinationAddress = useBridgeStore(
+    (state) => state.setDestinationAddress,
+  );
+  const isAddressInputError = useBridgeStore(
+    (state) => state.isAddressInputError,
+  );
+  const setIsAddressInputError = useBridgeStore(
+    (state) => state.setIsAddressInputError,
+  );
+  const setSendingAmount = useBridgeStore((state) => state.setSendingAmount);
+  const setReceivingAmount = useBridgeStore(
+    (state) => state.setReceivingAmount,
+  );
+
+  const {
+    status: isSourceChainModalOpen,
+    open: openSourceChainModal,
+    close: closeSourceChainModal,
+  } = useModal(false);
+  const {
+    status: isDestinationChainModalOpen,
+    open: openDestinationChainModal,
+    close: closeDestinationChainModal,
+  } = useModal(false);
+  const {
+    status: isTokenModalOpen,
+    open: openTokenModal,
+    close: closeTokenModal,
+  } = useModal(false);
+  const {
+    status: isConfirmBridgeModalOpen,
+    open: openConfirmBridgeModal,
+    close: closeConfirmBridgeModal,
+  } = useModal(false);
+
+  const sourceTypedChainId = useMemo(() => {
+    return calculateTypedChainId(
+      selectedSourceChain.chainType,
+      selectedSourceChain.id,
+    );
+  }, [selectedSourceChain]);
+
+  const destinationTypedChainId = useMemo(() => {
+    return calculateTypedChainId(
+      selectedDestinationChain.chainType,
+      selectedDestinationChain.id,
+    );
+  }, [selectedDestinationChain]);
+
+  const routerQuoteParams: RouterQuoteProps = useMemo(() => {
+    const fromTokenAddress =
+      sourceTypedChainId === PresetTypedChainId.TangleMainnetEVM
+        ? ROUTER_NATIVE_TOKEN_ADDRESS
+        : selectedToken.address;
+
+    const toTokenAddress =
+      destinationTypedChainId === PresetTypedChainId.TangleMainnetEVM
+        ? ROUTER_NATIVE_TOKEN_ADDRESS
+        : selectedToken.address;
+
+    const routerQuoteParams = {
+      fromTokenAddress,
+      toTokenAddress,
+      amountInWei: amount?.toString() ?? '',
+      fromTokenChainId: selectedSourceChain.id.toString(),
+      toTokenChainId: selectedDestinationChain.id.toString(),
+    };
+
+    return routerQuoteParams;
+  }, [
+    amount,
+    sourceTypedChainId,
+    selectedToken.address,
+    destinationTypedChainId,
+    selectedSourceChain.id,
+    selectedDestinationChain.id,
+  ]);
+
+  const hyperlaneQuoteParams: HyperlaneQuoteProps = useMemo(() => {
+    return {
+      token: selectedToken,
+      amount: Number(amount?.toString() ?? '0'),
+      sourceTypedChainId,
+      destinationTypedChainId,
+      senderAddress: activeAccount?.address ?? '',
+      recipientAddress: destinationAddress ?? '',
```

Comment:
What will happen when the amount is zero or the addresses are empty strings? Should we be handling these edge cases explicitly instead?